mm2src/coins/lp_coins.rs

```diff
@@ -2707,23 +2739,103 @@ impl Default for PrivKeyActivationPolicy {
     fn default() -> Self { PrivKeyActivationPolicy::ContextPrivKey }
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub enum PrivKeyPolicy<T> {
-    KeyPair(T),
+    Iguana(T),
+    HDWallet {
+        /// Derivation path of the coin.
+        /// This derivation path consists of `purpose` and `coin_type` only
+        /// where the full `BIP44` address has the following structure:
+        /// `m/purpose'/coin_type'`.
+        derivation_path: StandardHDPathToCoin,
+        activated_key: T,
+        bip39_secp_priv_key: ExtendedPrivateKey<secp256k1::SecretKey>,
+    },
     Trezor,
+    #[cfg(target_arch = "wasm32")]
+    Metamask(EthMetamaskPolicy),
```

Comment:
I had the same thought when moving this but I believe this can be extended to other extensions in the future (e.g. kepler) so I moved it here until it is changed. I would like to keep it like that for now since this is not used yet.

> ```rust
> #[cfg(target_arch = "wasm32")]
>     Metamask(T),
> ````

For this suggestion, metamask is only for EVM coins, so this still specific to EVM.